x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/search/aggregations/MergedGeoLinesTests.java

```diff
@@ -76,4 +99,191 @@ public void testMergeWithEmptyGeoLine() {
         assertArrayEquals(sortedValues, mergedGeoLines.getFinalSortValues(), 0d);
         assertArrayEquals(sortedPoints, mergedGeoLines.getFinalPoints());
     }
+
+    public void testMergeNonOverlappingGeoLinesAppendOnly() {
```

Comment:
It is not clear to me how do you test the case where you have the line split among two data nodes.